packages/truffle-debugger/lib/session/sagas/index.js

```diff
@@ -15,98 +15,104 @@ import * as web3 from "lib/web3/sagas";
   debug("waiting for contract information");
   let { contexts, sources } = yield take(actions.RECORD_CONTRACTS);
 
   debug("recording contract binaries");
-  yield *recordContexts(...contexts);
+  yield* recordContexts(...contexts);
 
   debug("recording contract sources");
-  yield *recordSources(...sources);
+  yield* recordSources(...sources);
 
   debug("waiting for start");
   // wait for start signal
-  let {txHash, provider} = yield take(actions.START);
+  let { txHash, provider } = yield take(actions.START);
   debug("starting");
 
   // process transaction
   debug("fetching transaction info");
-  let err = yield *fetchTx(txHash, provider);
+  let err = yield* fetchTx(txHash, provider);
   if (err) {
     debug("error %o", err);
-    yield *error(err);
-
+    yield* error(err);
   } else {
     debug("visiting ASTs");
     // visit asts
-    yield *ast.visitAll();
+    yield* ast.visitAll();
 
     debug("readying");
     // signal that stepping can begin
-    yield *ready();
+    yield* ready();
   }
 }
 
 export default prefixName("session", saga);
 
-
-function *forkListeners() {
+function* forkListeners() {
   return yield all(
-    [ast, controller, data, evm, solidity, trace, web3]
-      .map( app => fork(app.saga) )
+    [ast, controller, data, evm, solidity, trace, web3].map(app =>
+      fork(app.saga)
+    )
   );
 }
 
 function* fetchTx(txHash, provider) {
-  let result = yield *web3.inspectTransaction(txHash, provider);
+  let result = yield* web3.inspectTransaction(txHash, provider);
 
   if (result.error) {
     return result.error;
   }
 
-  yield *evm.begin(result);
+  yield* evm.begin(result);
 
-  let addresses = yield *trace.processTrace(result.trace);
+  let addresses = yield* trace.processTrace(result.trace);
   if (result.address && addresses.indexOf(result.address) == -1) {
     addresses.push(result.address);
   }
 
-  let binaries = yield *web3.obtainBinaries(addresses);
+  let binaries = yield* web3.obtainBinaries(addresses);
 
   yield all(
-    addresses.map( (address, i) => call(recordInstance, address, binaries[i]) )
+    addresses.map((address, i) => call(recordInstance, address, binaries[i]))
   );
 }
 
 function* recordContexts(...contexts) {
   for (let { contractName, binary, sourceMap } of contexts) {
-    yield *evm.addContext(contractName, { binary });
+    yield* evm.addContext(contractName, { binary });
 
     if (sourceMap) {
-      yield *solidity.addSourceMap(binary, sourceMap);
+      yield* solidity.addSourceMap(binary, sourceMap);
     }
   }
 }
 
 function* recordSources(...sources) {
-  for (let { sourcePath, source, ast } of sources) {
-    yield *solidity.addSource(source, sourcePath, ast);
+  for (let i = 0; i < sources.length; i++) {
+    const sourceData = sources[i];
+    if (sourceData !== undefined && sourceData !== null) {
+      yield* solidity.addSource(
+        sourceData.source,
+        sourceData.sourcePath,
+        sourceData.ast
+      );
+    }
```

Comment:
But why is it undefined?

Reply:
Something to do with interfaces I believe. When importing something goes wrong with them. I dont know. Saw it somewhere as a fix and there was no fix merged anywhere so I created a pull request.

Reply:
(Btw, I'm guessing most of these nonfunctional changes due to that linter we're now using?  Man, this is going to make reviewing PR's a bit of a pain for a while...)

Reply:
After looking deeper at the code... I'm still pretty confused as to why this is occurring!  I could maybe see it if the `files` option was used, but it doesn't seem like the cases causing the problem involve that.